experimenter/experimenter/kinto/tests/test_tasks.py

```diff
@@ -749,6 +752,78 @@ def test_launching_experiment_live_when_record_is_in_main(self):
             ).exists()
         )
 
+    def test_launching_experiment_live_prefflips_feature(self):
+        prefflips_feature = NimbusFeatureConfigFactory.create(
+            application=NimbusExperiment.Application.DESKTOP,
+            name="prefFlips",
+            slug="prefFlips",
+        )
+        launching_experiment = NimbusExperimentFactory.create_with_lifecycle(
+            NimbusExperimentFactory.Lifecycles.LAUNCH_APPROVE,
+            application=NimbusExperiment.Application.DESKTOP,
+            feature_configs=[prefflips_feature],
+        )
+
+        self.setup_kinto_get_main_records([])
+        self.setup_kinto_no_pending_review()
+
+        # Checking the desktop collection should not affect this recipe.
+        tasks.nimbus_check_kinto_push_queue_by_collection(
+            settings.KINTO_COLLECTION_NIMBUS_DESKTOP
+        )
+        self.mock_push_task.assert_not_called()
+
+        # Checking the secure collection should publish this recipe.
+        tasks.nimbus_check_kinto_push_queue_by_collection(
+            settings.KINTO_COLLECTION_NIMBUS_SECURE
+        )
+        self.mock_push_task.assert_called_once_with(
+            settings.KINTO_COLLECTION_NIMBUS_SECURE, launching_experiment.id
+        )
+
+    def test_checking_experiment_prefflips_rejection(self):
```

Comment:
In theory there's a lot of cases to test here, ie all the cases in the sequence diagrams.  Like we need to be able to recover all the intermediate aborted states like rejections and timeouts, where the collection needs to be rolled back.  Should we parameterize all the test cases in here and then assert them all on pref flip feature > secure collection?  And there's three broad cases to consider, ie desktop non pref flip feature > desktop collection, desktop with pref flip feature > secure collection, desktop with both non pref flip and pref flip feature > secure collection, all three of those combinations across every kinto interaction.  What we don't want is some particular kinto interaction ie timing out on a live update on a rollout (as a random example) calls into the wrong collection.

Reply:
> desktop with both non pref flip and pref flip feature > secure collection
no that cant be published. you cant have a multifeature prefFlips experiment with "non secure" features.

Reply:
Ah okay so then just the other two cases, but my broader point was there are a lot of sequences to keep intact with this change that I think it would be prudent to cover in testing?